documentation/src/main/asciidoc/manual-index.asciidoc

```diff
@@ -361,3 +361,226 @@ parameters through - see <<lucene-indexing-performance>> for more details.
+configure different parameters in order to bootstrap the mass-indexing job.
+
+==== Define your indexing scope
+Our job allows you to define your own indexing scope—you can start a full indexation or a partial
+indexation through 3 different methods: selecting the desired entity types, using HQL, or using
+Hibernate criteria.
+
+.How to add define the indexing scope in different methods?
+====
+[source, JAVA]
+----
+// full indexation
+Properties jobProps1 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .build();
+
+// partial indexation using HQL
+Properties jobProps2 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( "select c from MyClass c where c.name in ( 'Foo', 'Bar' )" )
+        .build();
+
+// partial indexation using HQL
+Properties jobProps3 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( Restrictions.in( "name", "Foo", "Bar" ) )
+        .build();
+----
+====
+
+==== Parallelism
+In order to maximize the performance, we highly recommend you to speed up the mass indexer using
+parallelism. Parallelism is activated by default. Under the standard JSR 352, the exact word is
+_"partitioning"_: The indexing step may run as multiple partitions, one per thread. Each partition
+has its own partition ID and parameters. If there’re more partitions than threads, partitions are\
+considered as a queue to consume: each thread can only run one partition at a time and won’t consume
+the next partition until the previous one is finished.
+
+===== Threads
+The maximum number of threads used by the job execution is defined through method `maxThreads()`.
+Within the N threads given, there’s 1 thread reserved for the core, so only N - 1 threads are
+available for different partitions. The default number of threads used in Hibernate Search is 10,
+which correspond to the max number of threads in JBeret. You can overwrite you maximum number used
+by Hibernate Search batch job, but we cannot guarantee that your batch runtime can provide the
+requested number.
+
+====
+[source, JAVA]
+----
+MassIndexingJob.parameters()
+        .maxThreads( 5 )
+        ...
+----
+====
+
+[NOTE]
+====
+Note that the batch runtime cannot guarantee the requested number of threads are available, it will
+use as it can up to the requested maximum. (JSR352 v1.0 Final Release, page 34)
+====
+
+This approach is optimized for best performance but requires to set the application in maintenance
+mode. Running a mass indexer in busy hours is not recommended. You also need to ensure the batch
+runtime has enough available threads for the execution, because all the batch jobs share the same
+thread pool.
+
+===== Row per partitions
+Mass indexer defines dynamically the number of partitions, but you need to tell it the number of
+rows per partitions so that it can calculate the partitions by division. Each partition deals with
```

Comment:
> Mass indexer defines dynamically the number of partitions [...] so that it can calculate the partitions by division

This feels like implementation details. Users don't care about how it's done internally, they just want to know how to use it.

What would you think of rephrasing the paragraph like this:

```
Each partition consists of a fixed number of elements to index.
You may tune exactly how many elements a partition will hold with `rowsPerPartition`.

[NOTE]
====
Each partition deals with one root entity type, so 2 different entity types will never run under the same partition.
====
```

Also, we should explain exactly how this number of rows per partition will influence the job. I mean, will it affect performance? Checkpoints? Something else? Why would setting this number to 10 instead of 10 million make a difference?